From 532977b339ee2d92f93a4a5d7716b65bb27a56d3 Mon Sep 17 00:00:00 2001
From: Hai Pham <hai.pham.ud@renesas.com>
Date: Mon, 26 May 2025 17:28:54 +0700
Subject: [PATCH 26/52] arm64: dts: ironhide: Add serial console pin control

Complete the descriptions of the serial console and the external serial
clock by adding pin control.

Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 arch/arm/dts/r8a78000-ironhide.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/dts/r8a78000-ironhide.dts b/arch/arm/dts/r8a78000-ironhide.dts
index d14f5ccf894..43c5aa308d5 100644
--- a/arch/arm/dts/r8a78000-ironhide.dts
+++ b/arch/arm/dts/r8a78000-ironhide.dts
@@ -76,10 +76,28 @@
 };
 
 &hscif0 {
+	pinctrl-0 = <&hscif0_pins>;
+	pinctrl-names = "default";
+
 	uart-has-rtscts;
 	status = "okay";
 };
 
+&pfc {
+	pinctrl-0 = <&scif_clk_pins>;
+	pinctrl-names = "default";
+
+	hscif0_pins: hscif0 {
+		groups = "hscif0_data", "hscif0_ctrl";
+		function = "hscif0";
+	};
+
+	scif_clk_pins: scif-clk {
+		groups = "scif_clk";
+		function = "scif_clk";
+	};
+};
+
 &scif_clk {
 	clock-frequency = <26000000>;
 };
-- 
2.34.1

